src/Solvers/ltdsolver.cpp

```diff
@@ -308,7 +308,7 @@ void LTDSolver::updateNodeQuality()
             if ( node->type() == Node::JUNCTION )
             {
                 // ... account for dilution from any external negative demand
-                if (node->outflow < 0.0 && node->qualSource == nullptr )
+                if (node->outflow < 0.0 && node->qualSource != nullptr )
```

Comment:
update the concentration if the inflow junction has Quality Source

Reply:
This change is not correct. The dilution occurs only for external inflows at nodes with no WQ source. For nodes with WQ sources the mixing of the external inflow with the node's current quality is handled in `QualSource::getQuality`.